group/assignment5/Assignment 5.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    private static void compareWords(String prevWord, String curWord, Node root, HashMap<Character, Node> visitedChars) {
+
+        boolean differenceFound = false;
+
+        int len = (prevWord.length() < curWord.length()) ? prevWord.length() : curWord.length();
+
+        int i = 0;
+        while (!differenceFound && (i < len)) {
+            char prev = prevWord.charAt(i);
+            char cur = curWord.charAt(i);
+            //if the chars are not the same, we should add dependency to the graph
+            if (prev != cur) {
+                differenceFound = true;
+                Node prevNode = null;
+                Node curNode = null;
+                //find parent node
+                //if it is the first appearance of the char, make new node for it
+                if (!visitedChars.containsKey(prev)) {
+                    prevNode = new Node(prev);
+                    root.addChild(prevNode);
+                    prevNode.incNumOfParents();
+                    visitedChars.put(prev, prevNode);
+                } else {
+                    prevNode = visitedChars.get(prev);
+                }
+                //find child node
+                //if it is the first appearance of the char, make new node for it
+                if (!visitedChars.containsKey(cur)) {
+                    curNode = new Node(cur);
+                    visitedChars.put(cur, curNode);
+                } else {
+                    curNode = visitedChars.get(cur);
+                }
+                //add dependency between nodes
+                curNode.incNumOfParents();
+                prevNode.addChild(curNode);
+            } else {
+                //if they are the same, make one Node instance so that the letter is in the alphabet
+                //if one does not exsist already
+                if (!visitedChars.containsKey(prev)) {
+                    Node newNode = new Node(prev);
+                    root.addChild(newNode);
+                    newNode.incNumOfParents();
+                    visitedChars.put(prev, newNode);
+                }
+            }
+            i++;
+        }
+        //add remaining charachters of both string, so that the alphabet is complete
+        for (int i1 = i; i1 < prevWord.length(); i1++) {
+            char newChar = prevWord.charAt(i1);
+            if (!visitedChars.containsKey(newChar)) {
+                Node newNode = new Node(newChar);
+                root.addChild(newNode);
+                newNode.incNumOfParents();
+                visitedChars.put(newChar, newNode);
+            }
+        }
+        for (int i1 = i; i1 < curWord.length(); i1++) {
+            char newChar = curWord.charAt(i1);
+            if (!visitedChars.containsKey(newChar)) {
+                Node newNode = new Node(newChar);
+                root.addChild(newNode);
+                newNode.incNumOfParents();
+                visitedChars.put(newChar, newNode);
+            }
+        }

```

Comment:
I agree with Ana on making some helper function, would be useful especially for these 2 repetitive for loops.